### ISSUE TITLE
Exception : You are trying to load a weight file containing System.String[] layers into a model with x layers

### ISSUE THREAD

**Issue by CrimyTheBold** [SciSharp/TensorFlow.NET#887]
As a model is not threadsafe, I'm trying to clone an existing model to use each clones in separate threads. I'm using save_weights + load_weights to recreate the clone (note that if you have another way to do it without using files, I'm in)

While loading the clone with load_weights I have the exception.
tensorflow.net 0.40.1
tensorflow.keras 0.5.1
scisharp.tensorflow.redist-windows-gpu 2.5.0

I reduced the problem to this unit test:

`

        public Functional BuildModel()
        {
            var inputs = keras.Input(shape: 2);

            // 1st dense layer
            var DenseLayer = keras.layers.Dense(1, activation: keras.activations.Sigmoid);
            var outputs = DenseLayer.Apply(inputs);

            // build keras model
            Functional model = keras.Model(inputs, outputs, name: Guid.NewGuid().ToString());
            // show model summary
            model.summary();

            // compile keras model into tensorflow's static graph
            model.compile(loss: keras.losses.MeanSquaredError(name: Guid.NewGuid().ToString()),
                optimizer: keras.optimizers.Adam(name: Guid.NewGuid().ToString()),
                metrics: new[] { "accuracy" });
            return model;
        }

        [Fact]
        public void Test1()
        {
            //Arrange
            string savefile = "mymodel.h5";
            var model1 = BuildModel();
            model1.save_weights(savefile);
            var model2 = BuildModel();

            //act
            model1.load_weights(savefile); //works
            model2.load_weights(savefile); //Exception : You are trying to load a weight file containing System.String[] layers into a model with 1 layers.

        }

        [Fact]
        public void Test2()
        {
            //Arrange
            string savefile = "mymodel2.h5";
            var model1 = BuildModel();
            model1.save_weights(savefile);
            model1 = BuildModel(); //recreate model

            //act
            model1.load_weights(savefile); //Exception : You are trying to load a weight file containing System.String[] layers into a model with 1 layers.

        }
`

**Comment by CrimyTheBold** [comment]
Just tested with 
tensorflow.net 0.60.5
tensorflow.keras 0.6.5
scisharp.tensorflow.redist-windows-gpu 2.6.0

same result

**Comment by Oceania2018** [comment]
@CrimyTheBold 
You can try to reset context before building a new model:
```csharp
using static Tensorflow.Binding;

tf.Context.reset_context();
var model1 = BuildModel();
model1.load_weights(savefile);
```
